core/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -2060,9 +2068,9 @@ public void startRecovery(RecoveryState recoveryState, PeerRecoveryTargetService
                 } else {
                     final RuntimeException e;
                     if (numShards == -1) {
-                        e = new IndexNotFoundException(mergeSourceIndex);
+                        e = new IndexNotFoundException(resizeSourceIndex);
                     } else {
-                        e = new IllegalStateException("not all shards from index " + mergeSourceIndex
+                        e = new IllegalStateException("not all shards from index " + resizeSourceIndex
```

Comment:
nit: "not all shards" -> "required shards of index" (now that we do splitting it may not be all)